olp-cpp-sdk-dataservice-read/src/generated/api/PlatformApi.cpp

```diff
@@ -51,24 +45,16 @@ client::CancellationToken PlatformApi::GetApis(
   std::string platform_url =
       "/platform/apis/" + service + "/" + service_version;
 
-  client::NetworkAsyncCallback network_callback =
-      [callback](client::HttpResponse response) {
-        if (response.status != 200) {
-          callback(ApisResponse(
-              client::ApiError(response.status, response.response.str())));
-        } else {
-          // parse the services
-          // TODO catch any exception and return as Error
-          callback(
-              ApisResponse(parser::parse<olp::dataservice::read::model::Apis>(
-                  response.response)));
-        }
-      };
-
-  return client.CallApi(platform_url, "GET", query_params, header_params,
-                        form_params, nullptr, "", network_callback);
+  auto response =
+      client.CallApi(platform_url, "GET", query_params, header_params,
+                     form_params, nullptr, "", context);
+  if (response.status != http::HttpStatusCode::OK) {
+    return ApisResponse(
+        client::ApiError(response.status, response.response.str()));
+  }
+  return ApisResponse(
+      parser::parse<olp::dataservice::read::model::Apis>(response.response));
```

Comment:
Should we not wrap this into the context.ExecuteOrCancelled() method so that we do not call anything in case it is already canceled?

Reply:
Why? The response contains an error when the request was canceled.